Implémentation du menu latéral et du Sac

fmkr-project opened the pull request:
Ajout d'un menu basique dans lequel il est possible de naviguer
Possibilité de ramasser des objets depuis la carte du monde
Ajout du Sac pour ranger ses objets, ainsi que d'objets variés séparés en catégories selon leur utilité

Note : le code est pour le moment relativement lourd et peu optimisé, il convient d'effectuer un (petit) nettoyage afin de le remettre aux normes... en particulier le fichier menu.

fmkr-project committed:
ébauche du Sac

Encore beaucoup de choses à fixer...

fmkr-project committed:
Amélioration graphique

L'objet disparaôt lorsqu'on le prend.
En attente de l'implémentation de la sauvegarde en BDD pour implémenter les flags relatifs aux objets

fmkr-project committed:
Merge branch 'main' into bag

fmkr-project committed:
Passage des touches en constantes

Je suggère d'enlever Ctrl droit en tant que touche de sprint (j'ai pas cette touche sur mon clavier), de toute façon une seule touche suffit

fmkr-project committed:
Hotfix : conflit d'affectation

fmkr-project committed:
Début d'implémentation du menu latéral

à suivre demain

fmkr-project committed:
Affichage du menu latéral

2 heures passées à résoudre un problème relativement immédiat :(

fmkr-project committed:
Début d'implémentat° des sous-menus

Sous-menus Sac et Sauvegarde
Encore un peu de cleaning à faire

fmkr-project committed:
Un menu fonctionnel

...avec deux options et qui clignote

fmkr-project committed:
Sous-menus ouvrables

Espace pour ouvrir, X pour fermer. Seul le Sac dispose d'une texture pour le moment, comme c'est la priorité de cette branche...

antoinech2 committed:
Squashing - Mise à jour avec les modification de main

antoinech2 committed:
Passage des controles en fichier de configuration.
Il n'est pas encore possible de les modifier dans le jeu

fmkr-project committed:
Affichage du nom et de la qté des objets

Encore quelques bugs d'affichage à régler. La flèche est aussi en cours d'implémentat°

fmkr-project committed:
Flèche dans le menu du Sac

La flèche apparaît même si l'inventaire est vide. On va supposer que l'inventaire n'est pas vide pour l'instant jusqu'à l'implémentation des groupes d'objets.
Reste maintenant à implémenter un menu basique d'interaction avec l'objet sélectionné, avant de commencer à merge le truc

fmkr-project committed:
Pages multiples & préparation au merge

fmkr-project committed:
Merge branch 'main' into bag

antoinech2 reviewed (commented):
https://user-images.githubusercontent.com/42295406/147077661-4e8478de-d70d-457d-aded-e07df72282d8.mp4

La flèche peut sortir du cadre de l'inventaire

fmkr-project commented:
Bug connu mais on suppose que l'inventaire n'est pas vide (cf. commit précédent, à voir en fonction de l'histoire)
Est-ce que le bug se produit avec des objets dans le Sac ?

antoinech2 committed:
Ajout d'un menu de début (v1) via F3
Affiche quelques informations utiles du jeu pour le débug

antoinech2 committed:
Ajout d'un menu de débug (v1) via F3
Affiche quelques informations utiles du jeu pour le débug

antoinech2 committed:
Merge branch 'bag' of https://github.com/antoinech2/prepa-simulator into bag

antoinech2 committed:
Ajout de la suppression rapide des données de config et sauvegarde.
Ctrl + Shift + F5 pour reset la config
Ctrl + Shift + F6 pour reset la sauvegarde

antoinech2 committed:
Cleaning : Suppression de la variable player.is_talking inutilisé

antoinech2 committed:
Passage de la variable menu_is_open dans la classe Game plutôt que Player

antoinech2 reviewed (approved):
C'est bon pour moi, il faudra voir à l'occasion pour améliorer l'affichage graphique (on dirait que le texte des objets dans le sac est superposé plein de fois).

Je vais m'occuper de la sauvegarde des objets

fmkr-project commented:
Pour le texte qui se superpose, c’est un problème récurrent avec le texte dans le jeu, il y a aussi de l’accumulation dans les dialogues

fmkr-project committed:
Hotfix : le menu lat ne se ferme plus

J'ai aussi mis le menu debug en rouge, car on n'y voit rien en I104

fmkr-project committed:
Derniers ajustements graphiques

Sauf bugs supplémentaires, le travail sur cette branche est fini. Prochaine étape : le scripting

antoinech2 commented:
Est-ce vraiment judicieux de lier le sac à la classe Player qui représente l'entité du joueur, en sachant qu'il sera à priori unique pour tout le jeu... A voir si ce n'est pas plus pratique dans la classe Game

fmkr-project commented:
D'un côté je trouve plus logique de lier Bag à Player, comme le sac appartient en soi au joueur et pas au jeu.
D'un autre côté :
- soit j'ai mal compris l'usage de la classe Player (qui apparemment est bourrée d'éléments graphiques uniquement)
- soit le sac est un élément unique au jeu, mais des problèmes peuvent apparaître si on commence à implémenter plusieurs données de sauvegarde

antoinech2 commented:
> D'un côté je trouve plus logique de lier Bag à Player, comme le sac appartient en soi au joueur et pas au jeu. D'un autre côté :
> 
> * soit j'ai mal compris l'usage de la classe Player (qui apparemment est bourrée d'éléments graphiques uniquement)
> * soit le sac est un élément unique au jeu, mais des problèmes peuvent apparaître si on commence à implémenter plusieurs données de sauvegarde

La classe Player est censé représenter le joueur dans le jeu (le personnage) et non pas le joueur IRL, ce qui explique qu'il n'y ait presque que du graphique. Alors que le sac va plus représenter l'inventaire du joueur IRL dans le jeu. Comme il est unique pour tout le jeu (il ne sera pas utile de faire un sac pour les NPC à priori), ça me semble plus simple de le lier à Game mais c'est à discuter...

En attendant je laisse lié au joueur pour la sauvegarde

fmkr-project committed:
Modification de la hiérarchie des classes

Finalement, assigner Bag à Game change vraiment peu le code, comme les seules fois où on doit l'appeler sont dans des lignes du style machin.game.player.bag.truc()

antoinech2 committed:
Sauvegarde du sac (bug connu lors de la redimension)

antoinech2 committed:
Bugfix : Erreur lors du chargement des données.
On introduit les données sous forme d'entier désormais.
Peut nécéssiter un reset sauvegarde

antoinech2 committed:
Bugfix : Possibilité de réinitialiser la sauvegarde via Ctrl + Shift +F6
Résolution d'une erreur Python

antoinech2 committed:
Bugfix : Impossible d'ouvrir le menu après une redimension.
Note : ne pas mettre des variables globales à une classe si elle 
dépendent d'un élément de pygame

fmkr-project committed:
I210 et 211, simplificat° de la BDD

L'identifiant d'une map est désormais plus intuitif : [numéro de la lettre][numéro de la salle] pour les salles numérotées, [numéro de la lettre, rien pour les étages 2 et 1][numéro de l'étage x 100] pour les paliers

fmkr-project committed:
Petites modifs graphiques

fmkr-project committed:
Ajout de commentaires
